cpp/models/ode_secir/parameters_io.h

```diff
@@ -271,133 +299,53 @@ IOResult<void> set_population_data(std::vector<Model<FP>>& model, const std::str
-                    scaling_factor_icu * num_icu[i] * mu_I_U[i][age] / sum_mu_I_U[i];
-                rki_data[i][t]((size_t)InfectionState::Recovered + (size_t)InfectionState::Count * age) =
-                    num_rec[i][age];
-                rki_data[i][t]((size_t)InfectionState::Dead + (size_t)InfectionState::Count * age) = num_death[i][age];
-                rki_data[i][t]((size_t)InfectionState::Susceptible + (size_t)InfectionState::Count * age) =
-                    num_population[i][age] - num_Exposed[i][age] - num_InfectedNoSymptoms[i][age] -
-                    num_InfectedNoSymptomsConfirmed[i][age] - num_InfectedSymptoms[i][age] -
-                    num_InfectedSymptomsConfirmed[i][age] - num_InfectedSevere[i][age] - num_rec[i][age] -
-                    num_death[i][age] -
-                    rki_data[i][t]((size_t)InfectionState::InfectedCritical + (size_t)InfectionState::Count * age);
-            }
+        BOOST_OUTCOME_TRY(details::set_confirmed_cases_data(models, confirmed_cases_path, case_data, region, offset_day,
+                                                            scaling_factor_inf));
+        BOOST_OUTCOME_TRY(details::set_population_data(models, num_population, region));
+        for (size_t r = 0; r < region.size(); r++) {
+            extrapolated_data[r][t] = models[r].get_initial_values();
         }
-        date = offset_date_by_days(date, 1);
     }
-    auto num_groups = (int)(size_t)model[0].parameters.get_num_groups();
-    BOOST_OUTCOME_TRY(save_result(rki_data, region, num_groups, path_join(dir, "Results_rki.h5")));
+    auto num_groups = (int)(size_t)models[0].parameters.get_num_groups();
+    BOOST_OUTCOME_TRY(save_result(extrapolated_data, region, num_groups, path_join(dir, "Results_rki.h5")));
 
-    auto rki_data_sum = sum_nodes(std::vector<std::vector<TimeSeries<double>>>{rki_data});
+    auto rki_data_sum = sum_nodes(std::vector<std::vector<TimeSeries<double>>>{extrapolated_data});
     BOOST_OUTCOME_TRY(save_result({rki_data_sum[0][0]}, {0}, num_groups, path_join(dir, "Results_rki_sum.h5")));
 
     return success();
 }
 #else
 template <class Model>
-IOResult<void> export_input_data_county_timeseries(std::vector<Model>&, const std::string&, std::vector<int> const&,
-                                                   Date, const std::vector<double>&, double, int, const std::string&,
-                                                   const std::string&, const std::string&)
+IOResult<void>
+export_input_data_county_timeseries(std::vector<Model> models, const std::string& dir, std::vector<int> const& region,
+                                    Date date, const std::vector<double>& scaling_factor_inf, double scaling_factor_icu,
+                                    int num_days, const std::string& divi_data_path,
+                                    const std::string& confirmed_cases_path, const std::string& population_data_path)
```

Comment:
I had also thought about this at first. However, this function is rarely used and it would make a functionality that is frequently used much more complex. We could certainly do this, but I'm ok with the current structure. The problem is the data that has to be read in. In order to do this efficiently for several iterations, we need a different structure so that the large files (casedata and population data) only have to be read in once.
The issue was more related to the many lines of code that were completely independent. These have been removed. Now we only use functions that have already been tested. This makes it easier to maintain and also clearer.

Reply:
Yes, i understand your argument